### PR TITLE
0.15 testing fixup

### PR DESCRIPTION
**Pull request by MCGallaspy**
This is WIP. Create expected level of isolation for behave tests. Basically, flush the database inbetween each scenario. Consequently, some tags which were applied to features need to be applied to each scenario.

Working now on individual test failures.

### CODE REVIEW

File: kalite/coachreports/features/environment.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 """
 environment.py specific to the this app
 """
-from kalite.testing.base_environment import before_all, after_all, before_feature, after_feature
+from kalite.testing.base_environment import *
```
Comment (benjaoming) [inline kalite/coachreports/features/environment.py]: This is usually not cool, any reason you want to import * ?

Comment (benjaoming) [inline kalite/coachreports/features/environment.py]: test

Comment (MCGallaspy) [inline kalite/coachreports/features/environment.py]: laziness?

Comment (MCGallaspy) [inline kalite/coachreports/features/environment.py]: But I will change it, since you are of course right

Comment (benjaoming) [inline kalite/coachreports/features/environment.py]: laziness can be a valid argument :)

Laziness is a good reason for PyCharm or LiClipse btw :)

File: kalitectl.py [hunk 2]
```diff
@@ -400,7 +400,7 @@ def start_watchify():
     sys.stdout.write('Starting watchify')
 
     watchify_process = subprocess.Popen(
-        ['node compile_javascript.js --debug --watch --staticfiles'],
+        args='node compile_javascript.js --debug --watch --staticfiles',
```
Comment (MCGallaspy) [inline kalitectl.py]: [This is recommended.](https://docs.python.org/2/library/subprocess.html#subprocess.Popen) Old version fails on Windows.

Comment (benjaoming) [inline kalitectl.py]: But Popen doesn't have a kwarg `args=...` ?

Comment (MCGallaspy) [inline kalitectl.py]: True, but it still works. e.g.
```python
>>> def f(a, b):
...     return a - b
...
>>> f(3,1)
2
>>> f(b=3,a=1)
-2
>>>
```

Comment (benjaoming) [inline kalitectl.py]: But is it necessary for it to work on Windows? Or was that the part where you changed the list to a string?

Comment (MCGallaspy) [inline kalitectl.py]: list to string. But it's still perfectly valid python. It seems [any argument can be referenced by name](https://docs.python.org/2/tutorial/controlflow.html#keyword-arguments), whether the argument is required or not (require==does not have a default value).

File: kalite/distributed/static/js/distributed/bundle_modules/homepage.js [hunk 3]
```diff
@@ -14,4 +14,12 @@ $(function() {
         }
     }
     window.statusModel.listenTo(window.statusModel, "change", content_rec_load);
+
+    window.statusModel.listenTo(window.statusModel, "change:has_superuser", function () {
+        if (!window.statusModel.get("has_superuser")) { // "has_superuser" is true if an admin user exists.
+            window.super_user_create_modal = new user.SuperUserCreateModalView();
+        } else if ( typeof window.super_user_create_modal !== "undefined" ) {
+            window.super_user_create_modal.remove();
+        }
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/bundle_modules/homepage.js]: @rtibbles is this kosher?

Comment (rtibbles) [inline kalite/distributed/static/js/distributed/bundle_modules/homepage.js]: As Christmas... I mean, yeah, looks fine.

File: kalite/coachreports/features/environment.py (hunk 1 above)
Comment (rtibbles) [inline kalite/coachreports/features/environment.py]: I think in this instance it is actually a good idea.

File: kalite/coachreports/features/steps/coachreports.py [hunk 4]
```diff
@@ -81,12 +81,40 @@ def impl(context):
 
 @given(u"I am on the tabular report")
 def impl(context):
+    create_some_learner_data()
```
Comment (rtibbles) [inline kalite/coachreports/features/steps/coachreports.py]: Not sure I am keen on this. Data should be ensured by other 'given' steps.

File: kalite/coachreports/features/steps/coachreports.py [hunk 5]
```diff
@@ -241,4 +269,4 @@ def impl(context):
 
 @when(u"I click on the completed colored cell")
 def impl(context):
-    click_and_wait_for_id_to_appear(context, find_css_with_wait(context, "td.complete[value=subtraction_1]"), "details-panel-view")
+    click_and_wait_for_id_to_appear(context, find_css_with_wait(context, "td.complete"), "details-panel-view")
```
Comment (rtibbles) [inline kalite/coachreports/features/steps/coachreports.py]: This specificity should be coming from a 'given' statement.

**Comment by rtibbles** [comment]
Merging. I will continue the test cleanup from develop.

Torch passed!

File: kalite/coachreports/features/environment.py (hunk 1 above)
Comment (benjaoming) [inline kalite/coachreports/features/environment.py]: @rtibbles `import *` is never a good idea when you are importing from a module that doesn't define `__all__`

Comment (rtibbles) [inline kalite/coachreports/features/environment.py]: In that case, I would say the problem here is that the module being imported from is missing an ```__all__```.

**Pull request merged**